client/containers/DashboardSubmissions/VerdictDialog.tsx

```diff
@@ -0,0 +1,154 @@
+				<p className="email-text-header">
+					<Icon icon="manually-entered-data" iconSize={12} />
+					{'  '}Email to Authors
+				</p>
+				<WorkflowTextEditor
+					initialContent={customEmailText}
+					onContent={setCustomEmailText}
+					placeholder="Specify message to pub author(s)."
+				/>
+			</div>
+			<div className={Classes.DIALOG_FOOTER}>
+				<div className={Classes.DIALOG_FOOTER_ACTIONS}>
+					<Checkbox
+						className="email-toggle"
+						checked={shouldSendEmail}
+						disabled={props.isHandlingSubmission}
+						onChange={(e) => {
+							setShouldSendEmail((e.target as HTMLInputElement).checked);
+						}}
+						label="Notify authors by email"
+					/>
+					<Button onClick={props.onClose} disabled={props.isHandlingSubmission}>
+						Cancel
+					</Button>
+					<Button
+						onClick={() => props.handleSubmission(customEmailText, shouldSendEmail)}
+						loading={props.isHandlingSubmission}
+						intent="primary"
+					>
+						Email & {props.actionTitle}
+					</Button>
+				</div>
+			</div>
+		</>
+	);
+};
+
+type PostSubmitBodyProps = {
+	completedName: string;
+	onClose: () => unknown;
+	isHandlingSubmission: boolean;
+};
+const PostSubmitBody = (props: PostSubmitBodyProps) => {
+	return (
+		<>
+			<div className={Classes.DIALOG_BODY}>
+				<Callout intent="success" title={`Submission ${props.completedName}!`}>
+					You successfully {props.completedName} the submission!
+				</Callout>
+			</div>
+			<div className={Classes.DIALOG_FOOTER}>
+				<div className={Classes.DIALOG_FOOTER_ACTIONS}>
+					<Button onClick={props.onClose} disabled={props.isHandlingSubmission}>
+						Close
+					</Button>
+				</div>
+			</div>
+		</>
+	);
+};
+
+const VerdictDialog = (props: Props) => {
+	const [isHandlingSubmission, setIsHandlingSubmission] = useState(false);
+	const [updatedSubmission, setUpdatedSubmission] = useState(null);
+	const [submissionError, setSubmissionError] = useState(null);
+	const handleSubmission = (customEmailText: DocJson, shouldSendEmail: boolean) => {
+		setIsHandlingSubmission(true);
+		apiFetch('/api/submissions', {
+			method: props.apiMethod,
+			body: JSON.stringify({
+				id: props.pub.submission.id,
+				skipEmail: !shouldSendEmail,
+				...(customEmailText && { customEmailText }),
+				...(props.status && { status: props.status }),
+			}),
+		})
+			.then((submissionRes) => {
+				setUpdatedSubmission(submissionRes);
+				setIsHandlingSubmission(false);
+			})
+			.then(() => props.onJudgePub(props.pub.id, props.status))
+			.catch((err) => {
+				setSubmissionError(err);
+				setIsHandlingSubmission(false);
+			});
+	};
+	return (
+		<Dialog
+			lazy={true}
+			title={props.actionTitle}
+			className="verdict-dialog"
+			isOpen={props.isOpen}
+			onClose={props.onClose}
+		>
+			{submissionError ? (
+				<Callout intent="warning" title="There was an error creating this submission." />
+			) : updatedSubmission ? (
+				<PostSubmitBody
```

Comment:
I can't get this to ever show up in testing. Under what circumstances does it appear?

Reply:
Good catch. Got some logic conflict here. You can get it to appear by starting in the 'all section' and then either approving or declining a submission. The conflict at play is that after changing a status, rather than letting the row languish in a filter view that ought to exclude it (e.g. a freshly 'declined' submission in the 'pending' view) it is omitted from the set of rows displayed. Since `VerdictDialog` is a child of the row itself, the success message is not dropped out of the dom together with it's parent row.